control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -2717,6 +2717,128 @@ func TestReconcileDeleteEndpoint(t *testing.T) {
 	}
 }
 
+// TestReconcileIgnoresServiceIgnoreLabel tests that the endpoints controller correctly ignores services
+// with the service-ignore label and deregisters services previously registered if the service-ignore
+// label is added. This is done by running Reconcile twice, once on an initial set of labels, then again
+// on a new set of labels with the service-ignore label. The number of services registered in Consul is
+// checked before updating the labels against the `svcsExpectedBeforeUpdate` and after updating the labels
+// where the expected number of services should be zero.
+func TestReconcileIgnoresServiceIgnoreLabel(t *testing.T) {
```

Comment:
This looks good, but I think it can be simplified so that we're not running `Reconcile()` twice, which could introduce weird behaviour in failure cases.

You can set it up to do a table test which holds the initial configuration of the pods, and services, and then an expected result, and then it runs with the following behaviour:
1. Initialize kube + consul with all of the initial configurations and service registrations.
2. Run `Reconcile()` 
3. Read Consul's services to see the expected end state has occurred.

Then you could run this with an initial configuration where the service is registered AND does/not have the label, and again when the service isn't registered AND does/not have the label.

Let me know if this does not make sense and I'd be happy to hop on a call and walk through it!

Reply:
Cool, I made that modification. Let me know what you think.